recipe/svtav1_300_support.patch

```diff
@@ -0,0 +1,81 @@
+From 4cb1a1a121a9e90a10079ea9511ff0f51676e7cd Mon Sep 17 00:00:00 2001
+From: Mark Harfouche <mark.harfouche@gmail.com>
+Date: Sun, 23 Feb 2025 19:03:27 -0500
+Subject: [PATCH] Backport of  Make code compatible with SVT-AV1 3.0 #2607
+
+---
+ cmake/Modules/LocalSvt.cmake |  6 ++++++
+ src/codec_svt.c              | 16 ++++++++++------
+ 2 files changed, 16 insertions(+), 6 deletions(-)
+
+diff --git a/cmake/Modules/LocalSvt.cmake b/cmake/Modules/LocalSvt.cmake
+index 72061349..0fc03eb0 100644
+--- a/cmake/Modules/LocalSvt.cmake
++++ b/cmake/Modules/LocalSvt.cmake
+@@ -57,6 +57,12 @@ else()
+     set(CMAKE_OUTPUT_DIRECTORY_ORIG "${CMAKE_OUTPUT_DIRECTORY}")
+     set(CMAKE_OUTPUT_DIRECTORY "${SVT_BINARY_DIR}" CACHE INTERNAL "")
+ 
++    if(CMAKE_INTERPROCEDURAL_OPTIMIZATION)
++        set(SVT_AV1_LTO ON)
++    else()
++        set(SVT_AV1_LTO OFF)
++    endif()
++
+     avif_fetchcontent_populate_cmake(svt)
+ 
+     set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE_ORIG} CACHE STRING "" FORCE)
+diff --git a/src/codec_svt.c b/src/codec_svt.c
+index 193f244a..40e53aa2 100644
+--- a/src/codec_svt.c
++++ b/src/codec_svt.c
+@@ -7,6 +7,7 @@
+ 
+ #include "svt-av1/EbSvtAv1Enc.h"
+ 
++#include <stdbool.h>
+ #include <stdint.h>
+ #include <string.h>
+ 
+@@ -114,7 +115,11 @@ static avifResult svtCodecEncodeImage(avifCodec * codec,
+         // See https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/1697.
+         memset(svt_config, 0, sizeof(EbSvtAv1EncConfiguration));
+ 
++#if SVT_AV1_CHECK_VERSION(3, 0, 0)
++        res = svt_av1_enc_init_handle(&codec->internal->svt_encoder, svt_config);
++#else
+         res = svt_av1_enc_init_handle(&codec->internal->svt_encoder, NULL, svt_config);
++#endif
+         if (res != EB_ErrorNone) {
+             goto cleanup;
+         }
+@@ -125,14 +130,13 @@ static avifResult svtCodecEncodeImage(avifCodec * codec,
+         svt_config->is_16bit_pipeline = image->depth > 8;
+ #endif
+ 
+-        // Follow comment in svt header: set if input is HDR10 BT2020 using SMPTE ST2084 (PQ).
+-        svt_config->high_dynamic_range_input = (image->depth == 10 && image->colorPrimaries == AVIF_COLOR_PRIMARIES_BT2020 &&
+-                                                image->transferCharacteristics == AVIF_TRANSFER_CHARACTERISTICS_SMPTE2084 &&
+-                                                image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_BT2020_NCL);
+-
```

Comment:
Just an FYI since I was confused about this seemingly breaking change. It seems that high_dynamic_range was never used in the public release of svt_avt1

https://github.com/AOMediaCodec/libavif/pull/2607#discussion_r1952809923